snowplow_analytics_sdk/json_shredder.py

```diff
@@ -18,25 +18,53 @@
 import json
 from snowplow_analytics_sdk.snowplow_event_transformation_exception import SnowplowEventTransformationException
 
+
+# TODO: remove in 0.3.0
 SCHEMA_PATTERN = re.compile(""".+:([a-zA-Z0-9_\.]+)/([a-zA-Z0-9_]+)/[^/]+/(.*)""")
 
+SCHEMA_URI = ("^iglu:"                        # Protocol
+              "([a-zA-Z0-9-_.]+)/"            # Vendor
+              "([a-zA-Z0-9-_]+)/"             # Name
+              "([a-zA-Z0-9-_]+)/"             # Format
+              "([1-9][0-9]*"                  # MODEL (cannot start with 0)
+              "(?:-(?:0|[1-9][0-9]*)){2})$")  # REVISION and ADDITION
 
-def fix_schema(prefix, schema):
+SCHEMA_URI_REGEX = re.compile(SCHEMA_URI)
+
+
+def extract_schema(uri):
```

Comment:
Please extract this code into a separate commit for https://github.com/snowplow/snowplow-python-analytics-sdk/issues/26